features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java

```diff
@@ -221,11 +236,81 @@ public void testNMS12740() throws Exception {
-            Assert.assertEquals(0, snmpInterfaceDao.findAllHavingFlows(2).size());
+            assertEquals(0, snmpInterfaceDao.findAllHavingFlows(2).size());
+        }
+    }
+
+    @Test
+    public void shouldDistinguishBetweenIngressAndEgressWhenDeterminingIfFlowsAreAvailable() throws Exception {
+        Assert.assertFalse(OnmsSnmpInterface.INGRESS_AND_EGRESS_REQUIRED);
+
+        stubFor(post("/_bulk")
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")));
+
+        final ClassificationEngine classificationEngine = new DefaultClassificationEngine(() -> Lists.newArrayList(
+                new RuleBuilder().withName("http").withDstPort("80").withProtocol("tcp,udp").build(),
+                new RuleBuilder().withName("https").withDstPort("443").withProtocol("tcp,udp").build()
+        ), FilterService.NOOP);
+
+        final DocumentEnricher documentEnricher = new DocumentEnricher(
+                new MetricRegistry(), nodeDao, interfaceToNodeCache, sessionUtils, classificationEngine,
+                new CacheConfigBuilder()
+                        .withName("flows.node")
+                        .withMaximumSize(1000)
+                        .withExpireAfterWrite(300)
+                        .build());
+
+        final JestClientFactory factory = new JestClientFactory();
+        factory.setHttpClientConfig(new HttpClientConfig.Builder("http://localhost:" + wireMockRule.port()).build());
+
+        try (JestClient client = factory.getObject()) {
+            final ElasticFlowRepository elasticFlowRepository = new ElasticFlowRepository(new MetricRegistry(),
+                    client, IndexStrategy.MONTHLY, documentEnricher,
+                    sessionUtils, nodeDao, snmpInterfaceDao,
+                    new MockIdentity(), new MockTracerRegistry(), new MockDocumentForwarder(), new IndexSettings(),
+                    mock(SmartQueryService.class));
+
+            Assert.assertThat(nodeDao.findAllHavingFlows(), is(empty()));
+            Assert.assertThat(snmpInterfaceDao.findAllHavingFlows(1), is(empty()));
+            Assert.assertThat(snmpInterfaceDao.findAllHavingFlows(2), is(empty()));
+
+            elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.EGRESS)), getMockFlowSource());
+            checkInterfaces(1, "192.168.1.1");
+            checkInterfaces(2);
+
+            elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.INGRESS)), getMockFlowSource());
+            checkInterfaces(1, "192.168.1.1", "192.168.1.1");
+            checkInterfaces(2);
+
+            elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.INGRESS)), getMockFlowSource());
+            checkInterfaces(1, "192.168.1.1", "192.168.1.1");
+            checkInterfaces(2);
```

Comment:
@fooker , @christianpape, the test will succeed with this implementation. But It is not clear to me why the system behaves like it does. Why is it green when I implement it that way?

I assume the checkInterfaces method is implemented wrong. I don't have enough domain knowledge to understand what we need to test for...
Maybe you can help me out here... Thanks!